hydromt/io.py

```diff
@@ -226,6 +228,81 @@ def open_mfraster(
     return ds
 
 
+def open_mfcsv(
+    fns: Dict[Union[str, int], Union[str, Path]],
+    concat_dim: str,
+    driver_kwargs: Optional[Dict[str, Any]] = None,
+    variable_axis: Literal[0, 1] = 1,
+) -> xr.Dataset:
+    """Open multiple csv files as single Dataset.
+
+    Arguments
+    ---------
+    fns : Dict[str | int, str | Path],
+        Dictionary containing a id -> filename mapping. Here the ids,
+        should correspond to the values of the `concat_dim` dimension.
+    concat_dim : str,
+        name of the dimention that will be created by concatinating
+        all of the supplied csv files.
+    driver_kwargs : Dict[str, Any],
+        Any additional arguments to be passed to pandas' `read_csv` function.
+    variable_axis : Literal[0, 1] = 1,
+        The axis along which your variables are. so if the csvs have the
+        columns as variable names, you would leave this as 1. If the variables
+        are along the index, set this to 0.
+
+    Returns
+    -------
+    data : Dataset
+        The newly created Dataset.
+    """
+    ds = xr.Dataset()
+    if variable_axis not in [0, 1]:
+        raise ValueError(f"there is no axis {variable_axis} available in 2D csv files")
+    # we're gonna use the structure of the first file found to check
+    # all others against
+    csv_kwargs = {"index_col": 0}
+    if driver_kwargs is not None:
+        csv_kwargs.update(**driver_kwargs)
+
+    first_id, first_fn = next(iter(fns.items()))
+    first_df = pd.read_csv(first_fn, **csv_kwargs)
+    if variable_axis == 0:
+        first_df = first_df.T
+
+    first_index = first_df.index
+
+    if first_df.index.name is None:
+        csv_index_name = "index"
+    else:
+        csv_index_name = first_df.index.name
+
+    first_df[concat_dim] = first_id
+    dfs = []
+    for id, fn in fns.items():
+        df = pd.read_csv(fn, **csv_kwargs)
+        if variable_axis == 0:
+            df = df.T
+
+        df[concat_dim] = id
+
+        if not df.index.dtype == first_index.dtype:
+            raise RuntimeError(
```

Comment:
Maybe more a ValueError or TypeError?

Reply:
yeah, I agree value error is a better one.